scalding-core/src/main/scala/com/twitter/scalding/CancellationHandler.scala

```diff
@@ -0,0 +1,37 @@
+package com.twitter.scalding
+
+import scala.concurrent.{ Future, ExecutionContext => ConcurrentExecutionContext, Promise }
+
+trait CancellationHandler { outer =>
+  def stop()(implicit ec: ConcurrentExecutionContext): Future[Unit]
+  def compose(other: CancellationHandler): CancellationHandler = new CancellationHandler {
+    override def stop()(implicit ec: ConcurrentExecutionContext): Future[Unit] = {
+      other.stop().flatMap(_ => outer.stop())
```

Comment:
I think we can do `Future.zip(other.stop(), outer.stop())` I don't think we need to cancel in order.

Reply:
👍